src/components/Form/FormFactory.js

```diff
@@ -200,19 +209,38 @@ const getFormGroups = ({ fields, fieldsValues, onFormChange, textPosition, label
       });
 
       let label;
-      if (horizontal && field.title) {
+
+      const labelFragment = (
+        <>
+          {field.type !== 'checkbox' && (
+            <React.Fragment>
+              <ControlLabel className={field.required ? 'required-pf' : null}>{field.title}</ControlLabel>
+              {field.help && (
+                <FieldLevelHelp className="kubevirt-form-group__field-help" placement="right" content={field.help} />
+              )}
+            </React.Fragment>
+          )}
+        </>
+      );
+
+      if (horizontal && !showLabels && field.title) {
         label = (
           <Col sm={labelSize} className={textPosition}>
-            {field.type !== 'checkbox' && (
-              <React.Fragment>
-                <ControlLabel className={field.required ? 'required-pf' : null}>{field.title}</ControlLabel>
-                {field.help && (
-                  <FieldLevelHelp className="kubevirt-form-group__field-help" placement="right" content={field.help} />
-                )}
-              </React.Fragment>
-            )}
+            {labelFragment}
           </Col>
         );
+
+        if (horizontal) {
```

Comment:
it seems this is always true, can you simplify the ifs which use `horizontal`?

Reply:
Lol, I must have been half asleep when I wrote this.